caffe2/core/context_base.h

```diff
@@ -1,7 +1,168 @@
 #pragma once
 
-#include <ATen/core/context_base.h>
-// For CaffeMap
+#include <array>
+#include <cstdlib>
+#include <ctime>
+#include <memory>
+#include <unordered_map>
+
+#include <c10/macros/Macros.h>
+#include <c10/core/Allocator.h>
+#include <c10/util/typeid.h>
+#include <c10/util/Exception.h>
+#include <c10/util/Registry.h>
+#include <c10/core/CopyBytes.h>
+
 #include "caffe2/core/common.h"
 #include "caffe2/core/logging.h"
 #include "caffe2/proto/caffe2_pb.h"
```

Comment:
Code movement only, except the headers that were previously in `caffe/core/context_base.h` are now inlined directly in the file.